src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs

```diff
@@ -685,14 +685,31 @@ internal IEnumerable<PortModel> GetGroupOutPorts(IEnumerable<NodeModel> ownerNod
                 );
         }
 
-        private List<ProxyPortViewModel> CreateProxyPorts(IEnumerable<PortModel> groupPortModels)
+        private List<PortViewModel> CreateProxyInPorts(IEnumerable<PortModel> groupPortModels)
         {
             var originalPortViewModels = WorkspaceViewModel.Nodes
-                .SelectMany(x => x.InPorts.Concat(x.OutPorts))
+                .SelectMany(x => x.InPorts)
+                .Where(x => groupPortModels.Contains(x.PortModel))
```

Comment:
Is this function called very often?

Reply:
I believe this is called on collapsing the group view correct?  @SHKnudsen

Reply:
Correct, its called every time the group is collapsed